core/graph/src/main/java/uk/gov/gchq/gaffer/graph/GraphConfig.java

```diff
@@ -116,25 +133,144 @@ public void setHooks(final List<GraphHook> hooks) {
         }
     }
 
+    /**
+     * Adds the supplied {@link GraphHook} to the list of graph hooks.
+     *
+     * @param hook The hook to add.
+     */
     public void addHook(final GraphHook hook) {
-        if (null != hook) {
-            if (hook instanceof GraphHookPath) {
-                final String path = ((GraphHookPath) hook).getPath();
-                final File file = new File(path);
-                if (!file.exists()) {
-                    throw new IllegalArgumentException("Unable to find graph hook file: " + path);
-                }
-                try {
-                    hooks.add(JSONSerialiser.deserialise(FileUtils.readFileToByteArray(file), GraphHook.class));
-                } catch (final IOException e) {
-                    throw new IllegalArgumentException("Unable to deserialise graph hook from file: " + path, e);
-                }
-            } else {
-                hooks.add(hook);
+        // Validate params
+        if (hook == null) {
+            LOGGER.warn("addHook was called with a null hook, ignoring call");
+            return;
+        }
+
+        if (hook instanceof GraphHookPath) {
+            final File file = new File(((GraphHookPath) hook).getPath());
+            if (!file.exists()) {
+                throw new IllegalArgumentException("Unable to find graph hook file: " + file.toString());
             }
+            try {
+                hooks.add(JSONSerialiser.deserialise(FileUtils.readFileToByteArray(file), GraphHook.class));
+            } catch (final IOException e) {
+                throw new IllegalArgumentException("Unable to deserialise graph hook from file: " + file.toString(), e);
+            }
+        } else {
+            LOGGER.debug("Adding Hook {}", hook.getClass().getSimpleName());
```

Comment:
Why log here but not when added from a path? I'd suggest also adding something there.

Reply:
Ahh this was something left from debugging the tests I'll apply to other one too